ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/LLMs/vllm/426__model_serving_vllm_metrics.robot

```diff
@@ -0,0 +1,114 @@
+    ...    inference_type=chat-completions    n_times=3    query_idx=8
+    ...    namespace=${TEST_NS}    string_check_only=${TRUE}
+
+Verify Vllm Metrics Are Present
+    [Documentation]    Confirm vLLM metrics are exposed in OpenShift metrics
+    [Tags]    Tier1    Sanity    Resources-GPU    RHOAIENG-6264
+    ${host} =    llm.Get KServe Inference Host Via CLI    isvc_name=vllm-gpt2-openai    namespace=${TEST_NS}
+    ${rc}    ${out}=    Run And Return Rc And Output
+    ...    curl -ks ${host}/metrics/
+    Should Be Equal As Integers    ${rc}    ${0}
+    Log    ${out}
+    ${thanos_url}=    Get OpenShift Thanos URL
+    ${token}=    Generate Thanos Token
+    Metrics Should Exist In UserWorkloadMonitoring    ${thanos_url}    ${token}    ${SEARCH_METRICS}
+
+
+*** Keywords ***
+Suite Setup
+    Skip If Component Is Not Enabled    kserve
+    RHOSi Setup
+    Set Default Storage Class In GCP    default=ssd-csi
+    ${is_self_managed}=    Is RHODS Self-Managed
+    IF    ${is_self_managed}
+        Configure User Workload Monitoring
+        Enable User Workload Monitoring
+        #TODO: Find reliable signal for UWM being ready
+        #Sleep    10m
+    END
+    Load Expected Responses
+
+Suite Teardown
+    Set Default Storage Class In GCP    default=standard-csi
+    ${rc}=    Run And Return Rc    oc delete inferenceservice -n ${TEST_NS} --all
+    Should Be Equal As Integers    ${rc}    ${0}
+    ${rc}=    Run And Return Rc    oc delete servingruntime -n ${TEST_NS} --all
+    Should Be Equal As Integers    ${rc}    ${0}
+    ${rc}=    Run And Return Rc    oc delete pod -n ${TEST_NS} --all
+    Should Be Equal As Integers    ${rc}    ${0}
+    ${rc}=    Run And Return Rc    oc delete namespace ${TEST_NS}
+    Should Be Equal As Integers    ${rc}    ${0}
+    RHOSi Teardown
+
+Set Default Storage Class In GCP
+    [Documentation]    If the storage class exists we can assume we are in GCP. We force ssd-csi to be the default class
+    ...    for the duration of this test suite.
+    [Arguments]    ${default}
+    ${rc}=    Run And Return Rc    oc get storageclass ${default}
+    IF    ${rc} == ${0}
+        IF    "${default}" == "ssd-csi"
+            Run    oc patch storageclass standard-csi -p '{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"false"}}}'  #robocop: disable
+            Run    oc patch storageclass ssd-csi -p '{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}'  #robocop: disable
+        ELSE
+            Run    oc patch storageclass ssd-csi -p '{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"false"}}}'  #robocop: disable
+            Run    oc patch storageclass standard-csi -p '{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}'  #robocop: disable
+        END
+    ELSE
+        Log    Proceeding with default storage class because we're not in GCP
+    END
```

Comment:
can you move this to a common place? it's present in `ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/LLMs/422__model_serving_llm_models.robot` as well